AK/NoAllocationGuard.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2022, kleines Filmröllchen <malu.bertsch@gmail.com>.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+// NOTE: The no-allocation guard relies on the LibC malloc implementation which checks the global s_allocation_enabled.
+//       Therefore, the guard doesn't work in the Kernel, but it has no use case there anyways.
+#if !defined(KERNEL)
```

Comment:
This seems kind of unfortunate, as it would be nice to be able to use this in kernel mode as well.

Reply:
I'm working on that right now, though it's not going too well

Reply:
I'm working on that right now, though it's not going too well